doc/api/process.md

```diff
@@ -2260,6 +2260,29 @@ process.kill(process.pid, 'SIGHUP');
 When `SIGUSR1` is received by a Node.js process, Node.js will start the
 debugger. See [Signal Events][].
 
+## `process.loadEnvFile(path)`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.1 - Active development
+
+* `path` {string | undefined}
+
+Loads the `.env` file into `process.env`. By default, the path is resolved to
+`path.resolve(process.cwd(), '.env')`. If file is not found, no error is thrown.
```

Comment:
I strongly disagree with this. What other Node APIs ignore being unable to find a requested file? Some of the `fs` APIs allow you to disable the error if the file doesn’t exist by passing `force: true`: https://nodejs.org/api/fs.html#fspromisesrmpath-options. I think either we allow error suppression via an option such as that, or we require the user to try/catch. We should never simply silently error or just warn.

Alternatively, if we change the input to be the string contents of the file rather than a file path, this issue goes away.

Reply:
There is an open issue around this. Even thought, try/catch would solve this issue, I think we should have the same behavior for both CLI and this API. Therefore, I recommend resolving https://github.com/nodejs/node/issues/51451 before resolving this `change of request`. What do you think @GeoffreyBooth?

Reply:
Sure, but I feel the same way there too. Are there any CLI flags that accept paths where we don’t error on invalid paths?

I don’t really care how dotenv behaves. What matters is that Node’s APIs are consistent and intuitive as a whole, for Node.

I also think it _would_ be better if this API just accepted the string contents of a `.env` file rather than a path to one. Getting the contents is trivial: a one-line `readFile` call. Then we don’t need to concern ourselves with what to do when the path doesn’t exist, and also this API doesn’t depend on how `--env-file` behaves because they have different inputs.

Reply:
> Sure, but I feel the same way there too. Are there any CLI flags that accept paths where we don’t error on invalid paths?

I'm not sure. I am interested in making the `Dotenv` to `--env-file` adoption easier for the users. I'm more inclined towards making it a warning, rather than an error.

> I don’t really care how dotenv behaves. What matters is that Node’s APIs are consistent and intuitive as a whole, for Node.

I agree. Therefore, not throwing an error in API, but in CLI might be inconsistent. Hence, I recommend behaving the same way.

There are some opinions around the error/warn in the issue by @benjamingr and @mcollina. Let's wait for a couple of days, and if we don't reach a consensus, we can bring it to the TSC meeting.

> I also think it would be better if this API just accepted the string contents of a .env file rather than a path to one. Getting the contents is trivial: a one-line readFile call.

In case you missed, I recommended an alternative API to support this use-case and also make the usage for the rest of the users, easy: https://github.com/nodejs/node/pull/51476#discussion_r1454079983 WDYT?

Reply:
> In case you missed, I recommended an alternative API to support this use-case and also make the usage for the rest of the users, easy: [#51476 (comment)](https://github.com/nodejs/node/pull/51476#discussion_r1454079983) WDYT?

I think there are two general options:

- A single API that does the common case by default, and that accepts an options object to cover other use cases (error/not error on missing file, load from file or string input, parse with/without setting on `process.env`, etc.)
- A very limited API that does _only the part that can’t be achieved via existing APIs,_ which is to parse string input into an object, and it’s on the user to get that input from somewhere (such as `readFile`) and do something with its output (such as set it on `process.env`).

The first option is simple _if you’re doing the default case._ But pass any of the options, or _multiple_ options, and now it’s approaching the complexity of doing each step yourself in two or three lines of code. The second option is simple if all you’re doing is parsing, which almost no one is, but you have control over the full process; it’s not like we need to continually add options to the “all in one” API for more use cases that we hadn’t thought of. The second option is more of a “small core” approach.

Reply:
To be clear - I am OK with failing/throwing on missing .env file - I was not expressing my opinion only the behavior of libraries (that typically don't throw). I'm also OK with not throwing.

This is one of those cases where we should consider developer experience over correctness since the feature's scope is limited and a part of its goal is to make peoples' lives easier.

So - not putting my weight behind either choice and will leave it to other people who spent more time thinking about this.